provisioning/miniprov/hare_mp/main.py

```diff
@@ -254,21 +255,202 @@ def test(args):
+    if cluster_sts:
+        logging.error('Cluster status reports failure')
+        resp = executecmds(['journalctl', '--since', timeinfo, '>', logfile])
+        logging.info('created journal log.%s', logfile)
+        exit(-1)
+
+    logging.info('PCS: Check all services are up.')
+    sleep(20)
+    resp = executecmds(pcs_status)
+    logging.info('PCS status: %s', resp[0])
+    if int(resp[2]):
+        logging.error('PCS failed to updated the status %s', resp[1])
+    else:
+        for line in resp[0]:
+            assert 'stopped' not in line, 'Some services are not up.' \
+                if 'Some services are not up.' else line
+
+    logging.info('hctl: Check that all the services are up.')
+    sleep(10)
+    resp = executecmds(hctl_status)
+    logging.info('hctl status: %s', resp[0])
+    if int(resp[2]):
+        logging.error('hctl failed to updated the status %s', resp[1])
+        resp = executecmds(['journalctl', '--since', timeinfo, '>', logfile])
+        logging.info('created journal log.%s', logfile)
+        raise Exception(
+            f'Command {hctl_status} exited with error code {int(resp[2])}.'
+            f'Command output: {resp[1]}')
+    else:
+        for line in resp[0]:
+            assert 'stopped' not in line, 'Some services are not up.' \
+                if 'Some services are not up.' else line
+    logging.info('Successfully performed cleanup after testing')
+
+
+# @pytest.mark.sanity
+def test_hare_bootstrap_shutdown(args):
+    """Test suite for single node hare init in loop."""
+    loop_count = int(args.dev[0])
+    hctl_status = ['hctl', 'status', '-d']
+    hctl_shutdown = ['hctl', 'shutdown']
+
+    test_hare_prereq()
+
+    resp = executecmds(hctl_status)
+    logging.info('hctl status: %s', resp[0])
+    if int(resp[2]):
+        logging.error('hctl failed to updated the status %s', resp[1])
+        raise Exception(
+            f'Command {hctl_status} exited with error code {int(resp[2])}.'
+            f'Command output: {resp[1]}')
+    else:
+        if 'Cluster is not running' not in resp[0]:
+            resp = executecmds(hctl_shutdown)
+            logging.info('hctl shutdown: %s', resp[0])
+
+        logging.info('-------Starting BOOTSTRAP-SHUTDOWN in LOOP-------')
+        for count in range(loop_count):
+            logging.info('Loop count# {}'.format(count + 1))
+            now = datetime.now()     # current date and time
+            date_time = now.strftime("%Y-%m-%d %H:%M:%S")
+            jlog = '~/journal_ctrl_' + now.strftime('%Y_%m_%d_%H%M%S') + '.log'
+
+            logging.info('Start hctl Bootstrap')
+            resp = bootstrap_cluster(str(args.file[0]), True)
+            if resp:
+                logging.error('Failed to bootstrap')
+                resp = executecmds(['journalctl', '--since', date_time, '>',
+                                    jlog])
+                logging.info('created journal log %s', jlog)
+                exit(-1)
```

Comment:
Please don't use `exit()`. Especially if this function will become a test runnable by pytest.

Reply:
Sure